### PR TITLE
Add error message to report

### PR DESCRIPTION
**Pull request by rhart**
## Description
Adds a new field, `message`, to the report function.

## Motivation and Context
Reporting the stackframes without the actual error message means you don't get the full context of the problem on the server you're reporting to.

## How Has This Been Tested?
Existing test has been modified.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] `node_modules/.bin/jscs -c .jscsrc stacktrace.js` passes without errors
- [x] `npm test` passes without errors
- [x] I have read the [contribution guidelines](CONTRIBUTING.md)
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes

### CODE REVIEW

File: stacktrace.js [hunk 1]
```diff
@@ -163,12 +163,12 @@
...
 
         /**
-         * Given an Array of StackFrames, serialize and POST to given URL.
+         * Given an error message and Array of StackFrames, serialize and POST to given URL.
          *
          * @param {Array} stackframes
          * @param {String} url
          */
-        report: function StackTrace$$report(stackframes, url) {
+        report: function StackTrace$$report(errorMsg, stackframes, url) {
```
Comment (eriwen) [inline stacktrace.js]: Putting the error message first breaks the API, I'm afraid.

Comment (rhart) [inline stacktrace.js]: Are you suggesting to move it to last?  Or is creating a completely separate method better?

Comment (eriwen) [inline stacktrace.js]: Moving the argument would be sufficient. We may also only want to assign the `message` if it's populated.

Comment (rhart) [inline stacktrace.js]: Done

**Comment by eriwen** [comment]
Thank you @rhart - This has been published `v1.2.0`

**Pull request merged**